ykrt/src/thread_intercept.rs

```diff
@@ -1,11 +1,54 @@
 use std::os::raw::{c_int, c_void};
 
+use std::{ffi::CString, ptr::null_mut};
+
+use libc::{dlsym, free, malloc, pthread_attr_t, pthread_create, pthread_t};
+
+const SHADOW_STACK_SIZE: usize = 1000000;
+
+#[derive(Debug)]
+struct ThreadRoutine {
+    pub args: *mut c_void,
+    pub start_routine: extern "C" fn(*mut c_void) -> *mut c_void,
+}
+
+extern "C" fn wrap_thread_routine(arg: *mut c_void) -> *mut c_void {
+    let str = CString::new("shadowstack_0").unwrap();
+    unsafe {
+        //  Obtain address of a shadowstack_0 symbol
+        let shadowstack_addr: *mut libc::c_void = dlsym(null_mut(), str.as_ptr() as *const i8);
+        if shadowstack_addr.is_null() {
+            panic!("Unable to find shadowstack address!")
```

Comment:
Nit-picking but in general we don't need "!" in error messages: the system exiting will induce sufficient additional adrenaline levels in the user :)

Reply:
removed 👉 d9ceb2e6d80eb0ced85945c289872a910074ddf4